packages/edit-post/src/components/sidebar/settings-header/index.js

```diff
@@ -1,65 +1,73 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { Button } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
-import { withDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+// translators: Label for the Block Settings Sidebar tab, not selected.
+const BLOCK_LABEL = __( 'Block' );
+// translators: Label for the Block Settings Sidebar tab, selected.
+const BLOCK_SELECTED_LABEL = __( 'Block (selected)' );
+
+const SettingsHeader = ( { sidebarName } ) => {
+	const { openGeneralSidebar } = useDispatch( 'core/edit-post' );
+
+	const documentLabel = useSelect( ( select ) => {
+		const currentPostType = select( 'core/editor' ).getCurrentPostType();
+		const postType = select( 'core' ).getPostType( currentPostType );
+
+		// translators: Default label for the Document sidebar tab, not selected.
+		const defaultDocumentLabel = __( 'Document' );
+
+		return get(
+			postType,
+			[ 'labels', 'singular_name' ],
+			defaultDocumentLabel
+		);
+	} );
 
-const SettingsHeader = ( {
-	openDocumentSettings,
-	openBlockSettings,
-	sidebarName,
-} ) => {
-	const blockLabel = __( 'Block' );
 	const [ documentAriaLabel, documentActiveClass ] =
 		sidebarName === 'edit-post/document'
 			? // translators: ARIA label for the Document sidebar tab, selected.
-			  [ __( 'Document (selected)' ), 'is-active' ]
-			: // translators: ARIA label for the Document sidebar tab, not selected.
-			  [ __( 'Document' ), '' ];
+			  [ sprintf( __( '%1$s (selected)' ), documentLabel ), 'is-active' ]
+			: [ documentLabel, '' ];
 
 	const [ blockAriaLabel, blockActiveClass ] =
 		sidebarName === 'edit-post/block'
-			? // translators: ARIA label for the Settings Sidebar tab, selected.
-			  [ __( 'Block (selected)' ), 'is-active' ]
-			: // translators: ARIA label for the Settings Sidebar tab, not selected.
-			  [ __( 'Block' ), '' ];
+			? [ BLOCK_SELECTED_LABEL, 'is-active' ]
+			: [ BLOCK_LABEL, '' ];
 
 	/* Use a list so screen readers will announce how many tabs there are. */
 	return (
 		<ul>
 			<li>
 				<Button
-					onClick={ openDocumentSettings }
+					onClick={ () => openGeneralSidebar( 'edit-post/document' ) }
```

Comment:
I'm not sure what our standards are for creating these functions inline, but it might be worth just bumping these to the main body of the component like:

```js
const openDocumentSettings = () => openGeneralSidebar( 'edit-post/document' );
const openBlockSettings = () => openGeneralSidebar( 'edit-post/block' );
```

Reply:
> I'm not sure what our standards

There are no standards for this, but inlining the function is very common nowadays. I would keep as it was, since it keeps everything that matters in one place, especially with such simple functions (they're basically a `Function#bind` call with a different syntax). But it's not a rule, either way would be fine.

Related reading: [React, Inline Functions, and Performance](https://reacttraining.com/blog/react-inline-functions-and-performance/) by Ryan Florence.

Reply:
Leaving performance out of the picture here, keeping

```js
const openDocumentSettings = () => openGeneralSidebar( 'edit-post/document' );
const openBlockSettings = () => openGeneralSidebar( 'edit-post/block' );
```

seems a bit more readable IMO :)

Reply:
It's up to y'all. :)

Reply:
14f0f0a

Reply:
Thanks for the link! Looks like an interesting read :)